src/yandex_cloud_ml_sdk/_datasets/dataset.py

```diff
@@ -161,14 +166,41 @@ async def _download(
         return await asyncio.wait_for(self.__download_impl(
             base_path=base_path,
             exist_ok=exist_ok,
+            timeout=timeout,
         ), timeout)
 
+    async def _read(
+        self,
+        *,
+        timeout: float,
```

Comment:
What this timeout means?
This parameter appears to be a global timeout for all reading. But in fact, it is not. Should we maybe set it for the entire dataset reading?

Reply:
Well, yeah, I know what you mean.
But it is already designed and used this way at all the places.
So, I reflected it in related shiny cool docstring